### PR TITLE
.NET 9preview5

### PR DESCRIPTION
**Pull request by vzarytovskii**
Upgrades SDK to 9p5

### CODE REVIEW

**Comment by vzarytovskii** [comment]
Might be related to https://github.com/dotnet/runtime/issues/102149, https://github.com/dotnet/runtime/issues/102149

**Comment by KevinRansom** [comment]
@vzarytovskii , I will take a look.

**Comment by vzarytovskii** [comment]
> @vzarytovskii , I will take a look.

No need for now, issue repros on C#. Trying to figure out if it's intentional.

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -9,16 +9,16 @@
       These will go away when repo updates targeting to net8.0
       Tracked with https://github.com/dotnet/fsharp/issues/14765
     -->
-    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/8.0.2" />
-    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Host.linux-x64/8.0.2" />
-    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Ref/8.0.2" />
+    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/8.0.4" />
```
Comment (MichaelSimons) [inline eng/SourceBuildPrebuiltBaseline.xml:12]: Consider defining these to be 8.0.* to reduce maintenance cost.

**Comment by vzarytovskii** [comment]
It's likely related to changes made in https://github.com/dotnet/runtime/pull/97529

**Comment by ViktorHofer** [comment]
@vzarytovskii just curious, will you also update to Arcade 9 soon? That would allow us to clean-up a bit of infra here.

**Comment by vzarytovskii** [comment]
> @vzarytovskii just curious, will you also update to Arcade 9 soon? That would allow us to clean-up a bit of infra here.

Yup, once we done with p5 upgrade, I will be moving us to net9 by default + consuming arcade and dotnet-optimization from the NET9 channels

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 2]
```diff
@@ -9,16 +9,16 @@
       These will go away when repo updates targeting to net8.0
       Tracked with https://github.com/dotnet/fsharp/issues/14765
     -->
-    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/8.0.2" />
-    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Host.linux-x64/8.0.2" />
-    <UsagePattern IdentityGlob="Microsoft.NETCore.App.Ref/8.0.2" />
+    <UsagePattern IdentityGlob="Microsoft.AspNetCore.App.Ref/8.0.*" />
```
Comment (KevinRansom) [inline eng/SourceBuildPrebuiltBaseline.xml:12]: Wild cards ... nice

File: tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModuleDynamic.fs [hunk 3]
```diff
@@ -94,7 +94,10 @@ module OperatorsModuleDynamic =
         
         // Overflow
         let result = Operators.byte Single.MaxValue
-        Assert.AreEqual(0uy, result)
+        if not Info.isNetFramework then
+            Assert.AreEqual(255uy, result)
+        else
+            Assert.AreEqual(0uy, result)
```
Comment (KevinRansom) [inline tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModuleDynamic.fs:100]: Why?

Comment (brianrourkeboll) [reply tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModuleDynamic.fs:100]: It probably wouldn't hurt to add a comment like this to this file:

```fsharp
// For more information on the runtime-dependent behavior differences in this file, see:
// - https://github.com/dotnet/runtime/issues/61885
// - https://github.com/dotnet/runtime/pull/97529
```

Comment (psfinaki) [reply tests/FSharp.Core.UnitTests/FSharp.Core/OperatorsModuleDynamic.fs:104]: Yep, good idea, added

**Review by psfinaki — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Comment by psfinaki** [comment]
@dotnet/source-build-internal please take a look.

**Comment by ViktorHofer** [comment]
Source-build changes LGTM

**Comment by ViktorHofer** [comment]
@MichaelSimons / @mthalman can you please approve?

**Review by mthalman — APPROVED** [review]

**Pull request merged**